src/lib.rs

```diff
@@ -1241,6 +1227,7 @@ impl<'a, L: 'static + Ledger> KeystoreState<'a, L> {
             .create(user_key.clone())?
             .with_description(description)
             .set_scan(scan)
+            .set_index(index)
```

Comment:
This is only ever set once per key, and is always set when we create a key, right? Wondering if `index` should be a required parameter to `create` rather than having a `set_index` function